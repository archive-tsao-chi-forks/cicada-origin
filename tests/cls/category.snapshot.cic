@def Category @the Type [
  Object: Type
  Morphism: (dom com: Object) -> Type
  id: (x: Object) -> Morphism(x)(x)

  // TODO

  // Object: Type
  // Morphism(dom: Object, cod: Object): Type
  // id(x: Object): Morphism(x, x)

  // compose(
  //   x y z: Object,
  //   f: Morphism(x, y),
  //   g: Morphism(y, z),
  // ): Morphism(x, z)

  // id_left(
  //   x y: Object,
  //   f: Morphism(x, y)
  // ): Equal(compose(x, x, y)(id(x), f), f)

  // id_right(
  //   x y: Object,
  //   f: Morphism(x, y),
  // ): Equal(compose(x, y, y)(f, id(y)), f)

  // compose_associative(
  //   x y z w: Object,
  //   f: Morphism(x, y),
  //   g: Morphism(y, z),
  //   h: Morphism(z, w),
  // ): Equal(
  //   compose(x, y, w)(f, compose(y, z, w)(g, h)),
  //   compose(x, z, w)(compose(x, y, z)(f, g), h))
]

@show Category
