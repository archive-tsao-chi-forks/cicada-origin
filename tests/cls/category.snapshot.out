@the Type [
  Object: Type
  Morphism: (com: Object) -> (dom: Object) -> Type
  id: (x: Object) -> Morphism(x)(x)
]

